lib/PublicShareAuth/Listener.php

```diff
@@ -112,8 +117,35 @@ public static function preventExtraGuestsFromJoining(Room $room): void {
 		}
 
 		$participantService = \OC::$server->get(ParticipantService::class);
-		$users = $participantService->getParticipantUserIds($room);
-		if ($room->getActiveGuests() > 0 || \count($users) > 1) {
+		if ($participantService->getNumberOfActors($room) > 1) {
+			throw new \OverflowException('Only the owner and another participant are allowed in rooms to request the password for a share');
+		}
+	}
+
+	/**
+	 * Prevents other users from being added to the room (as they will not be
+	 * able to join).
+	 *
+	 * This method should be called before a user is added to a room.
+	 *
+	 * @param Room $room
+	 * @param array[] $participants
+	 * @throws \OverflowException
+	 */
+	public static function preventExtraUsersFromBeingAdded(Room $room, array $participants): void {
+		if ($room->getObjectType() !== 'share:password') {
+			return;
+		}
+
+		// Events with more than one participant can be directly aborted, as
+		// when the owner is added during room creation or a user self-joins the
+		// event will always have just one participant.
+		if (count($participants) > 1) {
+			throw new \OverflowException('Only the owner and another participant are allowed in rooms to request the password for a share');
+		}
+
+		$participant = $participants[0];
```

Comment:
Check `!empty()` before? Maybe possible with the occ commands to trigger it otherwise (although it makes little sense)